### PR TITLE
apps/examples/power: Update power management test example

### PR DESCRIPTION
**Pull request by ewoodev**
Update power management test example
It is testing 3 case
1. chipset enter power-saving mode and wake-up
2. locking enter power-saving mode apis (pm_suspend/pm_resume)
3. scheduled timer wake-up source
4. uart wake-up source

### CODE REVIEW

File: apps/examples/power/power_main.c [hunk 1]
```diff
@@ -26,127 +26,172 @@
...
-	fd_lock = open(PM_LOCK_PATH, O_WRONLY);
-	if (fd_lock < 0) {
-		printf("Failed to open PM_LOCK file, ret: %d\n", fd_lock);
-		close(fd_unlock);
-		return NULL;
+
+	while (is_running) {
+		if (ioctl(fd, PMIOC_SLEEP, 100) < 0) {
+			printf("Fail to pm sleep(errno %d)\n", get_errno());
+			return -1;
```
Comment (sunghan-chang) [inline apps/examples/power/power_main.c:62]: Before leaving, you should close opened fd.

File: apps/examples/power/power_main.c [hunk 2]
```diff
@@ -26,127 +26,172 @@
...
-	if (ret < 0) {
-		printf("Failed to unlock PM state transition, ret: %d\n", ret);
-		goto errout_with_fd;
+static int pm_suspend_resume_test(void)
+{
+	printf("pm supend resume repeat test start\n");
+	pm_domain_arg_t domain_arg;
+	int domain_id;
+	int test_count = 0;
+	int	fd = open(PM_DRVPATH, O_WRONLY);
```
Comment (sunghan-chang) [inline apps/examples/power/power_main.c:75]: int`<tab>`fd -> int`<space>`fd

File: apps/examples/power/power_main.c [hunk 3]
```diff
@@ -26,127 +26,172 @@
...
-		printf("1st APP sleep complete\n\n");
+	domain_arg.domain_name = "SUSPEND_TEST";
+	if (ioctl(fd, PMIOC_DOMAIN_REGISTER, &domain_arg) < 0) {
+		printf("Fail to register pm domain(errno %d)", get_errno());
+		close(fd);
+		return -1;
+	}
+	domain_id = domain_arg.domain_id;
+
+	printf("%d domain id\n", domain_id);
```
Comment (sunghan-chang) [inline apps/examples/power/power_main.c:89]: Is it meaningful message? x domain id ? What can we get information from this? just value itself?

File: apps/examples/power/power_main.c [hunk 4]
```diff
@@ -26,127 +26,172 @@
...
 
-		printf("##########################################################\n\n");
-	} while (1);
+		printf("call resume(%d)\n", test_count);
+		if(ioctl(fd, PMIOC_RESUME, domain_id) < 0) {
+			printf("Fail to resume(errno %d)\n", get_errno());
+			break;
+		}
+
+		sleep(18);
```
Comment (edwakuwaku) [inline apps/examples/power/power_main.c:107]: Any reason to sleep(18)? It seems better to define a macro for duration instead of hardcoding it.

**Review by sunghan-chang — APPROVED** [review]

**Pull request merged**